src/main/java/com/hollingsworth/arsnouveau/common/block/EnchantingApparatusBlock.java

```diff
@@ -66,8 +71,20 @@ public ItemInteractionResult useItemOn(ItemStack stack, BlockState state, Level
                         colorPos.add(ColorPos.centeredAbove(pedPos));
                     }
                 }
+
                 Networking.sendToNearbyClient(world, tile.getBlockPos(), new HighlightAreaPacket(colorPos, 60));
-                PortUtil.sendMessage(player, Component.translatable("ars_nouveau.apparatus.norecipe"));
+
+                ImbuementTile imbuementTile = new ImbuementTile(pos, BlockRegistry.IMBUEMENT_BLOCK.defaultBlockState());
+                imbuementTile.setLevel(world);
+                imbuementTile.stack = player.getItemInHand(handIn).copy();
+                RecipeHolder<? extends IImbuementRecipe> imbue = imbuementTile.getRecipeNow();
```

Comment:
Making a fake tile to do these checks is going to end up being pretty fragile, I would extract the lookup to a static method in the tile if needed that takes the pedestal and input item instead, and just call that.

Reply:
resolved for IEnchantingRecipe, but not resolvable without breaking changes for IImbuementRecipe